src/Suggestor/EditorSuggestorPopup.ts

```diff
@@ -48,9 +53,20 @@ export class EditorSuggestor extends EditorSuggest<SuggestInfoWithContext> {
     getSuggestions(context: EditorSuggestContext): SuggestInfoWithContext[] {
         const line = context.query;
         const currentCursor = context.editor.getCursor();
+        const allTasks = this.plugin.getTasks();
+
```

Comment:
Picking an arbitrary location to track the following as a conversation...

There is a lot of new behaviour here that needs to be described to users, in the documentation.

I don't mind writing the docs, but I need all the behaviours to be recorded...

(this will also inform what automated tests need to be added for the new functionality...)

**Question: Are there any new behaviours that I have missed in the list below?**

### Existing docs to update

- Remove 'auto-suggest' from [limitations of task dependencies](https://publish.obsidian.md/tasks/Getting+Started/Task+Dependencies#Known+Limitations)
- Remove 'dependencies' from [limitations of auto-suggest](https://publish.obsidian.md/tasks/Editing/Auto-Suggest#Current+Limitations)
    - Which means removing that whole section - and that in turn means removing [this heading](https://publish.obsidian.md/tasks/Support+and+Help/Known+Limitations#Editing+Tasks+Auto-Suggest)
- Update [Editing dependencies](https://publish.obsidian.md/tasks/Getting+Started/Task+Dependencies#Editing+dependencies)
- Update [Auto suggest details](https://publish.obsidian.md/tasks/Editing/Auto-Suggest#Details)

### Things to note about new feature

- The new options added to the Auto Suggest menu are `Task ID` and `Task depends on ID`.
- When selecting `Task depends on ID`:
    - it searches descriptions of not-done tasks.
    - It initially filters to show tasks in the same file, but if you type more text, it will search all non-done tasks in the vault.
    - Unlike the Modal, it does not display the file names of candidate tasks.
- To depend on multiple tasks, type a comma after the last id in an existing 'depends on' value, and select another task.
- Do not create cyclic dependencies

Reply:
Sorry, I didn't have much Overview in the Docs.
With those Pointers, I can include them in the Next Commit.

> Question: Are there any new behaviours that I have missed in the list below?

Not too my Knowledge, Except:
> Unlike the Modal, it does not display the file names of candidate tasks.

If wanted, we probably could add this Behavior? 
Should i do so?

> Do not create cyclic dependencies
A Word of Warning might be more appropriate. 
You can create Cyclic Dependencies. BUT the Tasks will disappear from the "Not Blocked" Filter without a Warning.

Reply:
> Sorry, I didn't have much Overview in the Docs. With those Pointers, I can include them in the Next Commit.
> 
> > Question: Are there any new behaviours that I have missed in the list below?
> 
> Not too my Knowledge

Thanks

> Except:
> 
> > Unlike the Modal, it does not display the file names of candidate tasks.
> 
> If wanted, we probably could add this Behavior? Should i do so?

I think it's probably best not to add new functionality to this PR right now, in the interests of getting the feature released.

> > Do not create cyclic dependencies
> 
> A Word of Warning might be more appropriate. You can create Cyclic Dependencies. BUT the Tasks will disappear from the "Not Blocked" Filter without a Warning. And there is Currently now Way, of Showing Circular Dependecies.

I understand...

But it really helps to have PRs focussed on single issues, and the above comment is definitely a different issue from Auto Suggest...

Reply:
> I think it's probably best not to add new functionality to this PR right now, in the interests of getting the feature released.

Its as easy as Replacing the Existing Line with:
```
displayText: `${task.descriptionWithoutTags} - From: ${task.filename}.md`
````
Its Rendered as HTML Elemnt but cant get <i> or <em> Tags too work.
Ff you give me a Hint on how to Format it in Italic, i can Modify it too look as in the Modal.
I just need too know how you want it done.
For now its Included the Way it is above.

I noticed when Documenting the Feature, it might be Useful to some to define Custom ID's.
Currently, if you Select the ID Symbol, the ID has been Automatically Appended & Created.
I Changed the Behavior, that on Enter the ID Symbol gets Added, and then a new ID gets suggested.
The User now gets the chance to type his own ID instead if he wants too, or just hit Enter Again to Auto Generate a Unique ID. This should be a Quality of Life Improvment.

Reply:
Picking up these points:

- The docs
    - Thank you for updating the Task Dependencies docs
    - After merging the PR, I will add the detailed docs to the Auto Suggest page
- "displayText: `${task.descriptionWithoutTags} - From: ${task.filename}.md`"
    - I'll keep this change for now. I may later change it to be consistent with the Modal, which only shows the file name for tasks outside the file being edited.
    - But please don't do that now, I want to go ahead and merge the code pretty much as-is.
- "I Changed the Behavior, that on Enter the ID Symbol gets Added, and then a new ID gets suggested."
    - OK. I'll experiment that too, to test its usability in my own vault, after merging.